apisix/discovery/kubernetes/init.lua

```diff
@@ -615,4 +615,46 @@ function _M.init_worker()
     end
 end
 
+
+function _M.dump_data()
+
+    local eps = {}
+    for _, conf in ipairs(local_conf.discovery.kubernetes) do
+
+        local id = conf.id
+        local endpoint_dict = get_endpoint_dict(id)
+        local keys, err = endpoint_dict:get_keys()
+        if err then
+            error(err)
+            break
+        end
+
+        if keys then
+            local k8s = {}
+            for i = 1, #keys do
+
+                local key = keys[i]
+                --skip key with suffix #version
+                if key:sub(-#"#version") ~= "#version" then
```

Comment:
using `ngx.re.sub` is more performant.

Reply:
I think string.sub more  performant: It typically performs well due to its simplicity involving basic index calculation and memory copying.